apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -21,20 +37,31 @@ import { CardComponent } from '../../ui/card/card.component';
     `,
   ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [CardComponent, ListItemComponent],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TeacherCardComponent implements OnInit {
-  teachers: Teacher[] = [];
-  cardType = CardType.TEACHER;
+export class TeacherCardComponent {
+  private http = inject(FakeHttpService);
+  protected store = inject(TeacherStore);
+
+  constructor() {
+    const fetchTeachers = toSignal(this.http.fetchTeachers$, {
+      initialValue: [],
+    });
 
-  constructor(
-    private http: FakeHttpService,
-    private store: TeacherStore,
-  ) {}
+    effect(
+      () => {
+        this.store.addAll(fetchTeachers());
+      },
+      { allowSignalWrites: true },
+    );
+  }
```

Comment:
I'm not a big fan of this implementation. Using `allowSignalWrites` is often a code smell.

You can still use subscribe when needed. Using only signal is not the way to go. You should use signal when they fit, otherwise stay with rxjs. The goal is not to over-engineer your code